xDrip/Scenes/Settings/AlertSettings/SettingsAlertSingleType/SettingsAlertSingleTypePresenter.swift

```diff
@@ -22,7 +23,201 @@ final class SettingsAlertSingleTypePresenter: SettingsAlertSingleTypePresentatio
+                createRightSwitchCell(.snoozeFromNotification, isSwitchOn: config.snoozeFromNotification, switchValueChangedHandler: response.switchValueChangedHandler),
+                createCountDownPickerView(.defaultSnooze, detailText: "\(Int(config.defaultSnooze)) m", valueChangeHandler: response.timePickerValueChangedHandler),
+                createRightSwitchCell(.repeat, isSwitchOn: config.repeat, switchValueChangedHandler: response.switchValueChangedHandler),
+                createDisclosureCell(.sound, selectionHandler: response.selectionHandler),
+                createRightSwitchCell(.vibrate, isSwitchOn: config.isVibrating, switchValueChangedHandler: response.switchValueChangedHandler)
+            ])
+        }
+        
+        cells.append(
+            createRightSwitchCell(.entireDay, isSwitchOn: config.isEntireDay, switchValueChangedHandler: response.switchValueChangedHandler)
+        )
+        
+        if !config.isEntireDay {
+            let startTimeDate = Date(timeIntervalSince1970: config.startTime)
+            let endTimeDate = Date(timeIntervalSince1970: config.endTime)
+            let startTimeString = DateFormatter.localizedString(from: startTimeDate, dateStyle: .none, timeStyle: .short)
+            let endTimeString = DateFormatter.localizedString(from: endTimeDate, dateStyle: .none, timeStyle: .short)
+            
+            cells.append(contentsOf: [
+                createTimePickerView(.startTime, date: startTimeDate, detailText: startTimeString, valueChangeHandler: response.timePickerValueChangedHandler),
+                createTimePickerView(.endTime, date: endTimeDate, detailText: endTimeString, valueChangeHandler: response.timePickerValueChangedHandler)
+            ])
+        }
+        
+        let highTresholdString = User.current.settings.unit == .mmolL ? "\(config.highThreshold)" : "\(Int(config.highThreshold))"
+        let lowTresholdString = User.current.settings.unit == .mmolL ? "\(config.lowThreshold)" : "\(Int(config.lowThreshold))"
+        cells.append(contentsOf: [
+            createUnitsPickerView(.highTreshold, detailText: highTresholdString, valueChangeHandler: response.pickerViewValueChangedHandler),
+            createUnitsPickerView(.lowTreshold, detailText: lowTresholdString, valueChangeHandler: response.pickerViewValueChangedHandler)
+        ])
+        
+        return .normal(cells: cells, header: "settings_alert_single_type_events_header".localized, footer: nil)
+    }
+    
+    private func createTextInputViewCell(
+        _ field: SettingsAlertSingleType.Field,
+        detailText: String?,
+        placeholder: String?,
+        editingChangedHandler: @escaping (String) -> Void) -> BaseSettings.Cell {
+        
+        return .textInput(mainText: field.title, detailText: detailText, placeholder: placeholder) { (string) in
+            guard let string = string else { return }
```

Comment:
Why ignoring `nil`? It is a valid input as well and should be handled accordingly.